fe/fe-core/src/main/java/org/apache/doris/common/PatternMatcher.java

```diff
@@ -20,27 +20,58 @@
 import com.google.common.base.Strings;
 import com.google.common.collect.Sets;
 
+import java.util.Locale;
 import java.util.Set;
 import java.util.regex.Pattern;
 
 // Wrap for Java pattern and matcher
 public class PatternMatcher {
+    public static PatternMatcher MATCH_ANY = new PatternMatcher(Pattern.compile(".*"));
     private Pattern pattern;
+    private String casePattern;
```

Comment:
why use a `String` instead of a `Pattern`?

Reply:
String.equals() is faster than Pattern.matcher().matches(), and we can avoid errors caused by special characters, such as '.', '\', '*', etc. Of course, these special characters will not appear in the name, but it is better to use the string equivalent comparison.